pkg/apis/core/types_cloudprofile.go

```diff
@@ -269,9 +304,16 @@ const (
 // VersionClassification is the logical state of a version.
 type VersionClassification string
 
-// IsActive returns whether the version can be used.
-func (v VersionClassification) IsActive() bool {
-	return v != ClassificationExpired && v != ClassificationUnavailable
+var order = map[VersionClassification]int{
+	ClassificationUnavailable: 0,
+	ClassificationPreview:     1,
+	ClassificationSupported:   2,
+	ClassificationDeprecated:  3,
+	ClassificationExpired:     4,
+}
+
+func (c1 VersionClassification) Compare(c2 VersionClassification) int {
```

Comment:
Requires docstring. Example:
```suggestion
// Compare compares two VersionClassification objects to determine their order.
func (c1 VersionClassification) Compare(c2 VersionClassification) int {
```